### DISCUSSION TITLE
Not able to use initialized State from a function.

### DISCUSSION THREAD

**Discussion by kkroeger93** [clauderic/dnd-kit#303] [Q&A] [answered]
Currently I'm trying to get to understand, what the differences between this to State initializations are:

**Example 1:**
```
        const [items, setItems] = useState(() => {
            function createDataRow(id, name, quantity, unit, unitPrice, tax, discount, total) {
              return { id, name, quantity, unit, unitPrice, tax, discount, total };
            }

            let rows = [];
            rows.push(createDataRow(0, 'Lackieren', 5, 'Stk.', '120,00', 19, '-', '600,00'));
            rows.push(createDataRow(1, 'Montieren', 3, 'Stk.', '105,00', 19, '-', '315,00'));
            rows.push(createDataRow(2, 'Saugen', 3, 'Stk.', '105,00', 19, '-', '315,00'));
            return rows;
        });

````

**Example 2:**

```
        const [items, setItems] = useState(() => {
          return [
              { id: 1, name: 'Items One' },
              { id: 2, name: 'Item 2' },
              { id: 3, name: 'Items 3' },
            ];
        });
```

I'm asking this, because if I'm using the Example one, the State is somehow not mutable. 
And I'm getting errors when I'm trying to drag the Items around.

However, if I use the Example 2 my State is indeed mutable and I'm able to drag Items without problems.

### Example of my Test.

[![Edit sharp-paper-qbrrh](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/sharp-paper-qbrrh?fontsize=14&hidenavigation=1&theme=dark)

**Comment by clauderic** [comment, marked as answer]
They're both equivalent. The only difference is that in one of them you have an `id` that is `0` whereas the other ones start at `1`.

You can't safely use numbers for the `id` that you pass to `@dnd-kit`, they have to be strings. Refer to https://github.com/clauderic/dnd-kit/issues/286#issuecomment-848007168

https://codesandbox.io/s/festive-sara-nwh5c?file=/src/App.js

**Reply by kkroeger93** [reply]
Wow I feel so stupid right now..  I guess I was Code Blind. :)
Thank you very much.
